rotkehlchen/exchanges/bitstamp.py

```diff
@@ -246,15 +248,26 @@ def query_online_deposits_withdrawals(
             'sort': USER_TRANSACTION_SORTING_MODE,
             'offset': 0,
         }
+        history_db = DBHistoryEvents(self.db)
         if start_ts != Timestamp(0):
             # Get latest link from the DB to know where to resume from
             with self.db.conn.read_ctx() as cursor:
-                query_result = cursor.execute(
-                    'SELECT link FROM asset_movements WHERE location=? AND timestamp <= ? ORDER BY timestamp DESC LIMIT 1',  # noqa: E501
-                    (Location.BITSTAMP.serialize_for_db(), start_ts),
-                ).fetchone()
-                if query_result is not None:
-                    since_id = int(query_result[0]) + 1
+                query_result = history_db.get_history_events(
+                    cursor=cursor,
+                    filter_query=HistoryEventFilterQuery.make(
```

Comment:
If you want to use our own classes then you would need to also create an AssetMovementFilterQuery class, which would explicitly query asset movements. As this would return all types of events.

You can also simply mimic the approach they had before and do a direct DB query. Perhaps that's easier and it's definitely more performant.